SpatialGDK/Source/SpatialGDKTests/SpatialGDK/LoadBalancing/ReferenceCountedLockingPolicy/ReferenceCountedLockingPolicyTest.cpp

```diff
@@ -339,6 +361,339 @@ REFERENCECOUNTEDLOCKINGPOLICY_TEST(GIVEN_AcquireLock_and_ReleaseLock_are_called_
+
+	TSharedPtr<TestData> Data = MakeNewTestData(1, Worker_Authority::WORKER_AUTHORITY_AUTHORITATIVE, 1);
+
+	ADD_LATENT_AUTOMATION_COMMAND(FWaitForWorld(Data));
+
+	ADD_LATENT_AUTOMATION_COMMAND(FSpawnActor(Data, "Character"));
+	ADD_LATENT_AUTOMATION_COMMAND(FSpawnActor(Data, "Controller"));
+	ADD_LATENT_AUTOMATION_COMMAND(FSpawnActor(Data, "State"));
+
+	ADD_LATENT_AUTOMATION_COMMAND(FWaitForActor(Data, "Character"));
+	ADD_LATENT_AUTOMATION_COMMAND(FWaitForActor(Data, "Controller"));
+	ADD_LATENT_AUTOMATION_COMMAND(FWaitForActor(Data, "State"));
+
+	ADD_LATENT_AUTOMATION_COMMAND(FSetOwnership(Data, "State", "Controller"));
+	ADD_LATENT_AUTOMATION_COMMAND(FSetOwnership(Data, "Character", "Controller"));
+
+	ADD_LATENT_AUTOMATION_COMMAND(FTestIsLocked(this, Data, "Character", false));
+	ADD_LATENT_AUTOMATION_COMMAND(FTestIsLocked(this, Data, "Controller", false));
+	ADD_LATENT_AUTOMATION_COMMAND(FTestIsLocked(this, Data, "State", false));
+
+	ADD_LATENT_AUTOMATION_COMMAND(FAcquireLock(this, Data, "Controller", "First lock", true));
+
+	//		Controller (explicitly locked)
+	//		   /		 \
+	//		State     Character 
+
+	ADD_LATENT_AUTOMATION_COMMAND(FTestIsLocked(this, Data, "Character", true));
+	ADD_LATENT_AUTOMATION_COMMAND(FTestIsLocked(this, Data, "Controller", true));
+	ADD_LATENT_AUTOMATION_COMMAND(FTestIsLocked(this, Data, "State", true));
+
+	ADD_LATENT_AUTOMATION_COMMAND(FReleaseLock(this, Data, "Controller", "First lock", true));
+
+	//			Controller
+	//		   /		 \
+	//		State     Character 
+
+	ADD_LATENT_AUTOMATION_COMMAND(FTestIsLocked(this, Data, "Character", false));
+	ADD_LATENT_AUTOMATION_COMMAND(FTestIsLocked(this, Data, "Controller", false));
+	ADD_LATENT_AUTOMATION_COMMAND(FTestIsLocked(this, Data, "State", false));
+
+	return true;
+}
+
+REFERENCECOUNTEDLOCKINGPOLICY_TEST(GIVEN_AcquireLock_and_ReleaseLock_are_called_on_hierarchy_Actor_WHEN_Actor_hierarchy_changes_THEN_IsLocked_returns_correctly_for_all_Actors)
+{
+	AutomationOpenMap("/Engine/Maps/Entry");
+
+	TSharedPtr<TestData> Data = MakeNewTestData(1, Worker_Authority::WORKER_AUTHORITY_AUTHORITATIVE, 1);
+
+	ADD_LATENT_AUTOMATION_COMMAND(FWaitForWorld(Data));
+
+	ADD_LATENT_AUTOMATION_COMMAND(FSpawnActor(Data, "Character"));
+	ADD_LATENT_AUTOMATION_COMMAND(FSpawnActor(Data, "Controller"));
+	ADD_LATENT_AUTOMATION_COMMAND(FSpawnActor(Data, "State"));
+	ADD_LATENT_AUTOMATION_COMMAND(FSpawnActor(Data, "OtherActor"));
+
+	ADD_LATENT_AUTOMATION_COMMAND(FWaitForActor(Data, "Character"));
+	ADD_LATENT_AUTOMATION_COMMAND(FWaitForActor(Data, "Controller"));
+	ADD_LATENT_AUTOMATION_COMMAND(FWaitForActor(Data, "State"));
+	ADD_LATENT_AUTOMATION_COMMAND(FWaitForActor(Data, "OtherActor"));
+
+	ADD_LATENT_AUTOMATION_COMMAND(FSetOwnership(Data, "State", "Controller"));
+	ADD_LATENT_AUTOMATION_COMMAND(FSetOwnership(Data, "Character", "Controller"));
+
+	ADD_LATENT_AUTOMATION_COMMAND(FAcquireLock(this, Data, "Character", "First lock", true));
+
+	//			Controller			Other Actor
+	//			/		 \
```

Comment:
This might be a case to use spaces instead of tabs. Or find some other way to do this.